src/NuGet.Core/NuGet.Versioning/NuGetVersionFactory.cs

```diff
@@ -11,6 +11,10 @@ namespace NuGet.Versioning
 {
     public partial class NuGetVersion
     {
+        // Cached mappings from string => NuGetVersion. On cache hit, avoids allocations during TryParse.
+        private static Dictionary<string, NuGetVersion> ParsedNuGetVersionsMapping = new Dictionary<string, NuGetVersion>();
+        private const int ParsedNuGetVersionsMappingMaxEntries = 500;
```

Comment:
We know the implementation of `Dictionary` uses prime numbers for backing store capacity. Therefore, I think we should set max entries to the closest prime number to avoid "wasted space". If it's going to be allocated anyway, might as well use it.

Reply:
Done!

Reply:
Very open to whatever the consensus is here. (I personally like the original approach, as I don't see us hitting the 500 entry limit as likely, and it seems a bit less complex to understand)

My understanding of the original request was that if there ended up being between 501-521, we would end up vacating the cache when adding to it wouldn't have required it to resize. I don't know if that's exactly true, as the dictionary code is a bit more complex than that (they keep track of collisiion count when walking the buckets).

Reply:
From what I understand, the code ends up being identical if you specify 500 or 521. It uses the user specified capacity to set the initial bucket size, but it snaps the user specified capacity to the smallest prime that is greater than or equal to the specified capacity. I didn't intend for there to be any work as a result of my comment. Thought the background info would be interesting :)

Reply:
The code from the dictionary creation side is the same, whether that constant is 500 or 521. The code on my side that does the bounded size check is different based on that value. I'm just going to change back to 500 and remove the corresponding comment as it seems like more complication than that is worth.

Reply:
Oh right. I totally overlooked that. No wonder I was confused :)